plugin/globedit.vim

```diff
@@ -26,6 +26,10 @@ fun! globedit#run(cmd, pattern_list, ...) abort
 	let l:command = a:0 > 0 ? a:1 : ''
 
 	let l:errs = []
+
+	let l:save_autochdir = &autochdir
+	set noautochdir
```

Comment:
A better pattern for this is `try .. finally`:

```
let l:save_autochdir = &autochdir
try
    set noautochdir

    [.. rest of code ..]
finally
    let &autochdir = l:save_autochdir
endtry
```

The reason for this is that if the function error out somewhere half-way, then it won't execute the restoring of the setting and you're left with a Vim which has `autochdir` disabled. The `finally` block should always get run (except on syntax errors).

I'll just fix it after merge; sorry, I should have mentioned before. But just so you know for future reference :-)

Reply:
That makes sense. Might want to move the `set cpo&vim` inside the try as well then (although I'm still not sure what that does or why it's needed).

Reply:
It won't get executed at runtime, just once on startup: set `cpo`, parse the script`, reset `cpo`. It might break things on startup, but that's only really on parse errors which shouldn't happen in the first place (and not runtime errors like e.g. "invalid filename" or such, which would be a bug in the plugin btw, but bugs happen).

It's "needed" because `cpo` influences a lot of Vim behaviour, see `:help cpo`: if someone set some of these options in their vimrc then things in this script might break. There are all sorts of small and not-so-small things in there.